src/Integrations/Integrations/Laravel/LaravelIntegration.php

```diff
@@ -134,6 +150,13 @@ function ($This, $scope, $args) use ($rootSpan, $integration) {
             'fire',
             [
                 'prehook' => function (SpanData $span, $args) use ($integration) {
+                    if (
+                        dd_trace_env_config("DD_TRACE_REMOVE_AUTOINSTRUMENTATION_ORPHANS")
+                        && \DDTrace\get_priority_sampling() == DD_TRACE_PRIORITY_SAMPLING_AUTO_KEEP
```

Comment:
This is (everywhere) missing the check that the current span is indeed the root span. Otherwise it'll affect the whole trace. i.e. `&& $span == \DDTrace\root_span()`

Reply:
I think it would make most sense to not only check whether it's a local root span, but a trace root span, i.e. not part of distributed tracing either.

A better check would thus be `\DDTrace\trace_id() == $span->id`. This is only ever true for trace root spans.